### DISCUSSION TITLE
How to have a state (component) be hidden, then shows two different responses depending on button pressed?

### DISCUSSION THREAD

**Discussion by OSuwaidi** [reflex-dev/reflex#464] [Q&A]
Not sure if it’s a limitation of the project or a misunderstanding by me, but as of now, I do not see a way where I can have some state (component) be hidden from the enduser, and the state's response will only show up upon pressing (triggering) one of two buttons (True or False), where the response will be different for each button pressed.

A pythonic example would be:
```
if button_pressed is True:
	return “show_response_A"
elseif button_pressed is False:
	return “show_response_B"
else:  # Implies "button_pressed" is None (not pressed yet)
	return “"
```
I’m aware about the nested ```pc.cond``` component but I don’t see a way of implementing that logic with it.

So is there a way to accomplish the desired behaviour?

**Comment by Lendemor** [comment]
I would use a computed var:
```
@pc.var
def response(self):
    if self.button_pressed is True:
        return “show_response_A"
    elif self.button_pressed is False:
        return “show_response_B"
    else:  # Implies "button_pressed" is None (not pressed yet)
        return ""
```

**Reply by OSuwaidi** [reply]
If you set the class variable `button_pressed` to the base case, which is: `button_pressed = None`, you get the following error:
`pydantic.errors.ConfigError: unable to infer type for attribute "button_pressed"`, which I guess you can sidestep by doing: `button_pressed: bool`. 

However, is this the optimal way of implementing the desired behavior?

(P.S. How would you reset `button_pressed` to its base case (of returning "") when someone refreshes a page?)
